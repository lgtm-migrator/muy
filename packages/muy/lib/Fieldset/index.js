import React from "react"
import Box from "@material-ui/core/Box"
import clsx from "clsx"
import makeStyles from "@material-ui/core/styles/makeStyles"

/**
 * @todo Migrate some Fieldset styles to Box properties
 * @type {(props?: any) => ClassNameMap<string>}
 */
const useStyles = makeStyles({
    borderNone: {
        borderStyle: 'none'
    },
    borderSolid: {
        borderStyle: 'solid'
    }
})

const Fieldset = ({ className, variant, ...props }) => {
    const classes = useStyles()
    return (
        <Box
            border={variant === "borderSolid" ? 0 : 1}
            borderColor={variant === "borderSolid" && "divider"}
            borderStyle={variant === "borderSolid" ? "solid" : "none"}
            className={clsx(
                className,
                variant === "borderNone" && classes.borderNone,
                variant === "borderSolid" && classes.borderSolid
            )}
            component="fieldset"
            mb={3}
            mr={0}
            mt={0}
            pb={variant === "borderNone" ? 0 : 1.25}
            pr={variant === "borderNone" ? 0 : 1}
            pt={variant === "borderNone" ? 0 : 0.5}
            {...props}
        />
    )
}

export default Fieldset
