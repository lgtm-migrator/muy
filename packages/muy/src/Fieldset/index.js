import React from "react"
import Box from "@material-ui/core/Box"
import clsx from "clsx"
import makeStyles from "@material-ui/core/styles/makeStyles"

const useStyles = makeStyles(theme => ({
    root: {
        borderStyle: "none",
        borderWidth: 0,
        margin: `0 0 ${theme.spacing(3)}px`,
        padding: `${theme.spacing(0.5)}px ${theme.spacing(1)}px ${theme.spacing(
            1.25
        )}px`,
    },
    borderNone: {
        padding: 0,
    },
    borderSolid: {
        borderColor: theme.palette.divider,
        borderStyle: "solid",
        borderWidth: "1px",
    },
}))

const Fieldset = ({ variant, ...props }) => {
    const classes = useStyles()
    return (
        <Box
            className={clsx(
                classes.root,
                variant === "borderNone" && classes.borderNone,
                variant === "borderSolid" && classes.borderSolid
            )}
            component="fieldset"
            {...props}
        />
    )
}

export default Fieldset
