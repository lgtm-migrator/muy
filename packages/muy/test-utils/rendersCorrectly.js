import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
