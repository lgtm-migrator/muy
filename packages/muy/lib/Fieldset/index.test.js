import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import Fieldset from "../../lib/Fieldset"

describe("<Fieldset />", () => {
  it("renders correctly with no props", () => {
    rendersCorrectly(<Fieldset />)
  })
  it("renders correctly with borderNone variant", () => {
    rendersCorrectly(<Fieldset variant="borderNone" />)
  })
  it("renders correctly with borderSolid variant", () => {
    rendersCorrectly(<Fieldset variant="borderSolid" />)
  })
})
