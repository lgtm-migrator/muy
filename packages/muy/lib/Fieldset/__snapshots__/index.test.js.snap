// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fieldset /> renders correctly with borderNone variant 1`] = `
<fieldset
  className="MuiBox-root MuiBox-root-5 makeStyles-root-3 makeStyles-borderNone-1"
/>
`;

exports[`<Fieldset /> renders correctly with borderSolid variant 1`] = `
<fieldset
  className="MuiBox-root MuiBox-root-6 makeStyles-root-3 makeStyles-borderSolid-2"
/>
`;

exports[`<Fieldset /> renders correctly with no props 1`] = `
<fieldset
  className="MuiBox-root MuiBox-root-4 makeStyles-root-3"
/>
`;
